src/containers/tokens/components/bonding-curve-form/index.js

```diff
@@ -0,0 +1,422 @@
+import { connect } from 'react-redux'
+import { toBN } from 'ethjs'
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import { formValueSelector } from 'redux-form'
+
+import Button from '../../../../components/button'
+import { form } from '../../../../utils/form-generator'
+import { required, number, positiveNumber } from '../../../../utils/validation'
+import {
+  decimalStringToWeiBN,
+  weiBNToDecimalString
+} from '../../../../utils/number'
+import * as bondingCurveSelectors from '../../../../reducers/bonding-curve'
+import * as bondingCurveActions from '../../../../actions/bonding-curve'
+
+const {
+  Form: BuyPNKFromBondingCurveForm,
+  isInvalid: getBuyPNKFromBondingCurveFormIsInvalid,
+  submit: submitBuyPNKFromBondingCurveForm
+} = form('buyPNKFromBondingCurveForm', {
+  header: {
+    type: 'header',
+    props: { title: 'BUY PNK' }
+  },
+  amountOfETH: {
+    type: 'text',
+    validate: [required, number, positiveNumber],
+    props: {
+      type: 'number',
+      className: 'Form-noMargins'
+    }
+  },
+  rate: {
+    type: 'info',
+    props: {
+      className: 'Tokens-form-rate'
+    }
+  }
+})
+
+const { Form: SellPNKToBondingCurveFormHeader } = form(
```

Comment:
Why is the header a separate form?

Reply:
This is indeed a hack. I wanted to use create-redux-form to create the selling form but I also needed to put some custom `div`s into the form, which create-redux-form doesn't support. If I got rid of create-redux-form entirely and built upon redux-form, I would end up duplicate most of create-redux-form. So I split the form into two and wrapped `div` around the "sub-form".